backend/server/api/services/todos.service.js

```diff
@@ -18,6 +18,39 @@ export class ToDoService {
     const newTodo = allTodos.child(newTodoRef.key).toJSON();
     return { ...newTodo, key: newTodoRef.key };
   }
+
+  async search(searchTodo) {
+    let matchTodos = [];
+    await this.db.ref("todo").once(
+      "value",
+      function (snapshot) {
+        const todos = snapshot.val();
+        for (const property in todos) {
+          let todo = todos[property];
+          //If the respective filter is used and the fields match, then add that todo to the match array
+          if (searchTodo.useName && searchTodo.name === todo.name)
+            matchTodos.push(todo);
+          else if (
+            searchTodo.useDescription &&
+            searchTodo.description === todo.description
+          )
+            matchTodos.push(todo);
+          else if (searchTodo.useIsDone && searchTodo.isDone === todo.isDone)
+            matchTodos.push(todo);
+          else if (searchTodo.useDueDate && searchTodo.dueDate === todo.dueDate)
+            matchTodos.push(todo);
```

Comment:
You can leave out the `use*` and only check if the properties are an empty string

Reply:
Well ok, I'll assume that you cannot search for todos with empty description and dueDate. But I still believe that I need useIsDone, because of the boolean nature of isDone.